src/containers/Tenant/Schema/SchemaTree/SchemaTree.tsx

```diff
@@ -50,33 +59,80 @@ export function SchemaTree(props: SchemaTreeProps) {
         return childItems;
     };
 
+    React.useEffect(() => {
+        if (createDirectoryResult.isSuccess) {
+            setNavigationTreeKey(createDirectoryResult.requestId);
+            onActivePathUpdate(`${parent}/${child}`);
+            setCreateDirectoryOpen(false);
+        }
+    }, [createDirectoryResult, onActivePathUpdate, parent, child]);
```

Comment:
`createDirectory` returns promise, we can make these actions on promise resolve.